### PR TITLE
PersistenceTask

### PR DESCRIPTION
**Pull request by AlyaNovikova**

### CODE REVIEW

File: alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import Cocoa
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift:1]: Will Foundation be sufficient?

File: alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      todos = []
+    }
+  }
+
+  public func add(todo: Todo) throws {
+    todos.append(todo)
+    try todos.writeJSON(to: fileURL)
+  }
+
+  public func changeStatus(of index: Int, newStatus: Bool) throws -> Bool {
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift:31]: How about passing an item? 
You don't need Bool as the return type, do you?
Now there is two way of using API, one of which is incorrect:
```
let list = try TodoList()
var item = list.items[0]
item.isCompleted.swap()
```

File: alyanovikova/PersistenceTask/Tests/PersistenceTaskTests/PersistenceTaskTests.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    do {
+      let todos = [Todo(todo: "take a walk with dogs"), Todo(todo: "learn swift"), Todo(todo: "call mum")]
+      
+      let jsonEncoder = JSONEncoder()
+      let data = try jsonEncoder.encode(todos)
+      
+      try data.write(to: fileURL)
+    } catch {
+      XCTFail("Test failed with \(error)")
+    }
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Tests/PersistenceTaskTests/PersistenceTaskTests.swift:29]: I think you may omit this part

File: alyanovikova/PersistenceTask/Tests/PersistenceTaskTests/PersistenceTaskTests.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    // WHEN
+    let todoList = try TodoList()
+
+    // THEN
+    XCTAssert(FileManager.default.fileExists(atPath: fileURL.path))
+    XCTAssert(todoList.todos.count == 3)
+    XCTAssert(todoList.todos[0].todo == "take a walk with dogs" && todoList.todos[0].isCompleted == false)
+    XCTAssert(todoList.todos[1].todo == "learn swift" && todoList.todos[0].isCompleted == false)
+    XCTAssert(todoList.todos[2].todo == "call mum" && todoList.todos[0].isCompleted == false)
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Tests/PersistenceTaskTests/PersistenceTaskTests.swift:39]: Please use `XCTAssertEqual`, `XCTAssertFalse` throughout

File: alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import Cocoa
+
+public struct Todo: Codable {
+  public let todo: String
+  public var isCompleted: Bool
+
+  public init(todo: String, isCompleted: Bool = false) {
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift:7]: Nice idea with overloading!

**Comment by AlyaNovikova** [comment]
fixed!

File: alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+public struct Todo: Equatable, Codable {
+  public let todo: String
+  public var isCompleted: Bool
```
Comment (gartemiy) [inline alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift:5]: Don't you need to disallow updating isCompleted publicly?
I mean something like `public fileprivate(set) var isCompleted: Bool`

Comment (AlyaNovikova) [reply alyanovikova/PersistenceTask/Sources/PersistenceTask/PersistenceTask.swift]: Yes, true

**Pull request merged**